### PR TITLE
[Enhancement] md2yml pre-commit hook

### PR DESCRIPTION
**Pull request by xiexinch**
In this PR:

 We add a pre-commit hook to call `.dev/md2yml.py` at a new commit, which will:

- Update model metafiles according to the corresponding .md files.
- Update model-index.yml file according to the metafiles.
- Check if any file has been modified.

### CODE REVIEW

**Comment by Junjun2016** [comment]
Hi @xiexinch 
Please fix the lint error!

File: configs/ann/ann.yml [hunk 1]
```diff
@@ -0,0 +1,312 @@
...
+    backbone: R-50-D8
+    crop size: 512x1024
+    lr schd: 40000
+    inference time (ms/im):
+    - value: 269.54
+      hardware: V100
+      backend: PyTorch
+      batch size: 1
+      mode: FP32
+      resolution: (512x1024)
```
Comment (zhouzaida) [inline configs/ann/ann.yml:22]: The format of resolution is (h, w)

File: configs/apcnet/apcnet.yml [hunk 2]
```diff
@@ -0,0 +1,235 @@
...
+    Dataset: ADE20K
+    Metrics:
+      mIoU: 45.54
+      mIoU(ms+flip): 46.65
+  Config: configs/apcnet/apcnet_r101-d8_512x512_80k_ade20k.py
+  Weights: https://download.openmmlab.com/mmsegmentation/v0.5/apcnet/apcnet_r101-d8_512x512_80k_ade20k/apcnet_r101-d8_512x512_80k_ade20k_20201214_115704-c656c3fb.pth
+- Name: apcnet_r50-d8_512x512_160k_ade20k
+  In Collection: apcnet
+  Metadata:
+    method: APCNet
```
Comment (zhouzaida) [inline configs/apcnet/apcnet.yml:209]: whether the method filed is duplicate with In Collection

Comment (xiexinch) [reply configs/apcnet/apcnet.yml:209]: Is it illegal?

File: configs/ann/ann.yml [hunk 3]
```diff
@@ -0,0 +1,296 @@
...
+    Metrics:
+      mIoU: 41.01
+      mIoU(ms+flip): 42.3
+  Config: configs/ann/ann_r50-d8_512x512_80k_ade20k.py
+  Weights: https://download.openmmlab.com/mmsegmentation/v0.5/ann/ann_r50-d8_512x512_80k_ade20k/ann_r50-d8_512x512_80k_ade20k_20200615_014818-26f75e11.pth
+- Name: ann_r101-d8_512x512_80k_ade20k
+  In Collection: ann
+  Metadata:
+    backbone: R-101-D8
+    crop size: 512x512
```
Comment (zhouzaida) [inline configs/ann/ann.yml:179]: the format should be (h, w)

**Review by zhouzaida — APPROVED** [review]
LGTM

**Review by xvjiarui — APPROVED** [review]

**Pull request merged**